ops/c/src/mpi/ops_mpi_hdf5.cpp

```diff
@@ -286,6 +286,7 @@ void ops_fetch_block_hdf5_file(ops_block block, char const *file_name) {
     if (file_exist(file_name) == 0) {
       if (OPS_instance::getOPSInstance()->OPS_diags > 2)
         ops_printf("File %s does not exist .... creating file\n", file_name);
+      MPI_Barrier(OPS_MPI_HDF5_BLOCK_WORLD);
       if (ops_is_root()) {
```

Comment:
I don't quite remember why exactly we do this below, probably for performance reasons. However, with multi-block, this check is not right, because process 0 may not own part of this block. So this should be `if (my_rank==0)`